assets/backport/github/github.go

```diff
@@ -0,0 +1,355 @@
+}
+
+// cherryPickCommitsOnBranch cherry picks a list of commits on a given branch.
+func (c *Client) cherryPickCommitsOnBranch(ctx context.Context, branchName string, commits []string) error {
+	branch, _, err := c.Client.Repositories.GetBranch(ctx, c.Organization, c.Repository, branchName, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Get the branch's HEAD. 
+	headCommit, _, err := c.Client.Git.GetCommit(ctx,
+		c.Organization,
+		c.Repository,
+		*branch.Commit.SHA)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for i := 0; i < len(commits); i++ {
+		cherryCommit, _, err := c.Client.Git.GetCommit(ctx,
+			c.Organization,
+			c.Repository,
+			commits[i])
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		tree, sha, err := c.cherryPickCommit(ctx, branchName, cherryCommit, headCommit)
+		if err != nil {
+			defer func() {
+				refName := fmt.Sprintf("%s%s", branchRefPrefix, branchName)
+				c.Client.Git.DeleteRef(ctx, c.Organization, c.Repository, refName)
+			}()
+			return trace.Wrap(err)
+		}
+		headCommit.SHA = &sha
+		headCommit.Tree = tree
+	}
+	return nil
+}
+
+// cherryPickCommit cherry picks a single commit on a branch.
+func (c *Client) cherryPickCommit(ctx context.Context, branchName string, cherryCommit *go_github.Commit, headBranchCommit *go_github.Commit) (*go_github.Tree, string, error) {
+	if len(cherryCommit.Parents) != 1 {
+		return nil, "", trace.BadParameter("merge commits are not supported")
+	}
+	cherryParent := cherryCommit.Parents[0]
+	// Temporarily set the parent of the branch HEAD to the parent of the commit
+	// to cherry-pick so they are siblings. 
+	err := c.createSiblingCommit(ctx, branchName, headBranchCommit, cherryParent)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	// When git performs the merge, it detects that the parent of the branch commit that is 
+	// being merged onto matches the parent of the cherry pick commit, and merges a tree of size 1. 
+	// The merge commit will contain the delta between the file tree in target branch and the 
+	// commit to cherry-pick. 
+	merge, err := c.merge(ctx, branchName, *cherryCommit.SHA)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	mergeTree := merge.GetTree()
+
+	updatedCommit, _, err := c.Client.Git.GetCommit(ctx,
+		c.Organization,
+		c.Repository,
+		*headBranchCommit.SHA)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	// Create the actual cherry-pick commit on the target branch containing the merge commit tree. 
+	commit, _, err := c.Client.Git.CreateCommit(ctx, c.Organization, c.Repository, &go_github.Commit{
+		Message: cherryCommit.Message,
+		Tree:    mergeTree,
+		Parents: []*go_github.Commit{
+			updatedCommit,
+		},
+	})
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+
+	// Overwrite the merge commit and its parent on the branch by the newly created commit.
+	// The result will be equivalent to what would have happened with a fast-forward merge.
+	sha := commit.GetSHA()
```

Comment:
Below is invalid, so that is why I am assigning it to `sha`.  You can't get the memory address from a method that returns a value. 

```
	_, _, err = c.Client.Git.UpdateRef(ctx, c.Organization, c.Repository, &go_github.Reference{
		Ref: &refName,
		Object: &go_github.GitObject{
			SHA: &commit.GetSHA(),
		},
	}, true)
```